### PR TITLE
libbeat/processors/add_process_metada: add capabilities to process me…

### PR DESCRIPTION
**Pull request by haesbaert**
## Proposed commit message

Extends process metadata with effective and permitted capabilities.

Errors from `capabilities.FromPid()` are ignored since it returns a nil slice, which results in len() == 0 which supresses any output. A possible common error is getting ESRCH as the process might have already exited.

## Checklist

- [ ] ~~My code follows the style guidelines of this project~~
- [x]  I have commented my code, particularly in hard-to-understand areas
- [ ] ~~I have made corresponding changes to the documentation~~
- [ ] ~~I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

```
processors:
  - add_process_metadata:
      overwrite_keys: true
      match_pids: ["process.pid"]

auditbeat.modules:
- module: system
  datasets:
    - socket
  period: 10s # The frequency at which the datasets check for changes
  state.period: 20s

output.console:
  pretty: true
```

## Related issues

Part of https://github.com/elastic/security-team/issues/4375
Related to https://github.com/elastic/beats/pull/37453

## Screenshots

Screenshot from 2024-03-11 12-07-56

[output.txt](https://github.com/elastic/beats/files/14558386/output.txt)

### CODE REVIEW

File: libbeat/processors/add_process_metadata/add_process_metadata_test.go [hunk 1]
```diff
@@ -40,6 +41,10 @@ import (
 func TestAddProcessMetadata(t *testing.T) {
 	logp.TestingSetup(logp.WithSelectors(processorName))
 
+	capMock, err := capabilities.FromUint64(0xabacabb)
+	if err != nil {
+		t.Fatal(err)
```
[the remark this answers is not in the document]
Comment (haesbaert) [reply libbeat/processors/add_process_metadata/add_process_metadata_test.go:46]: Fine by me, the rest of the file just `Fatal(err)` though, was trying to be consistent.

**Comment by haesbaert** [comment]
> BuildKite should not prevent the PR from be merged yet. Once you get the approvals and Jenkins is green you should be able to merge it.

Awesome, thanks :)

**Review by stanek-michal — APPROVED** [review]
LGTM

**Review by nicholasberlin — APPROVED** [review]

**Comment by haesbaert** [comment]
Merging is blocked as waiting a review from @fearful-symmetry , is this because we need a reviewer from each team?

File: libbeat/processors/add_process_metadata/gosysinfo_provider.go [hunk 2]
```diff
@@ -52,17 +53,22 @@ func (p gosysinfoProvider) GetProcessMetadata(pid int) (result *processMetadata,
 		}
 	}
 
+	capPermitted, _ := capabilities.FromPid(capabilities.Permitted, pid)
+	capEffective, _ := capabilities.FromPid(capabilities.Effective, pid)
```
Comment (fearful-symmetry) [inline libbeat/processors/add_process_metadata/gosysinfo_provider.go:57]: Is there a reason why we're not checking the errors here?

Comment (haesbaert) [reply libbeat/processors/add_process_metadata/gosysinfo_provider.go:57]: If it fails, we don't include the capabilities, same outcome if the capabilities were empty. There are valid/normal reasons for it to fail, like the process being gone by the time we try to fetch it. Also on Windows/OS X i fails with ErrUnsupported.

Comment (fearful-symmetry) [reply libbeat/processors/add_process_metadata/gosysinfo_provider.go:57]: Can we add a quick comment to that effect?

Comment (haesbaert) [reply libbeat/processors/add_process_metadata/gosysinfo_provider.go:60]: Done.

**Review by fearful-symmetry — APPROVED** [review]

**Review by efd6 — APPROVED** [review]

**Comment by gwsales** [comment]
Because these capabilities increases the event size when doing multiple parent process lookups. Is it possible to add a flag to enable to disable this specific field for the lookup:

something like this:
```
processors:
  - add_process_metadata:
      match_pids:
        - process.pid
      capabilities: false
```
The desire is to disable adding this field when needed, but I'm okay with a default include.

**Pull request merged**